src/hooks/useLocation.js

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
+import RNLocation from 'react-native-location';
+
+import { INITIAL_COORDS, DISTANCE_FILTER } from 'constants/common';
+import parseError from 'utils/parseError';
+
+RNLocation.configure({
+  distanceFilter: DISTANCE_FILTER,
+});
+
+const useLocation = () => {
+  const [location, setLocation] = useState(INITIAL_COORDS);
+
+  useEffect(() => {
+    async function requestLocation() {
+      try {
+        let granted = await RNLocation.checkPermission({
+          ios: 'whenInUse',
+          android: {
+            detail: 'coarse',
+          },
```

Comment:
You are reusing this object twice, what do you think about having this object on a constant "permissions" and use it then